internal/watcher/minio.go

```diff
@@ -81,42 +81,52 @@ func NewMinioWatcher(ctx context.Context, logger logr.Logger, config *MinioConfi
 	}, nil
 }
 
-func (w *minioWatcher) Watch(ctx context.Context) (*BlobEvent, error) {
-	event, err := w.pop(ctx)
+func (w *minioWatcher) Watch(ctx context.Context) (*BlobEvent, Cleanup, error) {
+	event, val, err := w.pop(ctx)
+	cleanup := w.rem(val)
 	if err != nil {
 		if errors.Is(err, redis.Nil) {
 			err = ErrWatchTimeout
 		}
-		return nil, err
+		return nil, noopCleanup, err
 	}
 	if event.Bucket != w.bucket {
-		return nil, ErrBucketMismatch
+		return nil, noopCleanup, ErrBucketMismatch
 	}
-	return event, nil
+	return event, cleanup, nil
 }
 
 func (w *minioWatcher) Path() string {
 	return ""
 }
 
-func (w *minioWatcher) pop(ctx context.Context) (*BlobEvent, error) {
+// rem return a function that allows items from the failed list to be safely removed
+// in the event of a workflow or some other type of redis error.
+func (w *minioWatcher) rem(val string) func(ctx context.Context) error {
+	return func(ctx context.Context) error {
+		logger := w.logger.WithValues("list", w.failedList)
+		if _, err := w.client.LRem(ctx, w.failedList, 1, val).Result(); err != nil {
+			logger.Error(err, "Error removing message from failed list.", "list", w.failedList)
+			return err
+		}
+		logger.V(2).Info("Successfully removed messages.")
```

Comment:
Would it make more sense to use the singular form? i.e. `message`.

Reply:
got it. I thought you were talking about a method on logger for a second.

Reply:
I thought about this but there could be more than one message in the failed list right? We don't know how many actually live there so, messages makes sense. We could also do this: "Successfully removed message(s)."